test/unit-test/mqtt_agent_utest.c

```diff
@@ -81,14 +192,197 @@ int suiteTearDown( int numFailures )
 
 void test_MQTTAgent_Init_Happy_Path( void )
 {
+    MQTTAgentContext_t mqttAgentContext;
+    AgentMessageInterface_t msgInterface = { 0 };
+    MQTTFixedBuffer_t networkBuffer = { 0 };
+    TransportInterface_t transportInterface = { 0 };
+    IncomingPublishCallback_t incomingCallback;
+    void * incomingPacketContext;
+    AgentMessageContext_t msg;
+    MQTTStatus_t mqttStatus;
 
+    msgInterface.pMsgCtx = &msg;
+    msgInterface.send = mockSend;
+    msgInterface.recv = mockReceive;
+    msgInterface.releaseCommand = mockReleaseCommand;
+    msgInterface.getCommand = mockGetCommand;
 
+    MQTT_Init_ExpectAnyArgsAndReturn( MQTTSuccess );
+    mqttStatus = MQTTAgent_Init( &mqttAgentContext, &msgInterface, &networkBuffer, &transportInterface, getTime, mockPublishCallback, incomingPacketContext );
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    TEST_ASSERT_EQUAL_PTR( mockPublishCallback, mqttAgentContext.pIncomingCallback );
+    TEST_ASSERT_EQUAL_PTR( incomingPacketContext, mqttAgentContext.pIncomingCallbackContext );
+    TEST_ASSERT_EQUAL_MEMORY( &msgInterface, &mqttAgentContext.agentInterface, sizeof( msgInterface ) );
 }
 
 /**
  * @brief Test that any NULL parameter causes MQTTAgent_Init to return MQTTBadParameter.
  */
-void test_MQTT_Init_Invalid_Params( void )
+void test_MQTTAgent_Init_Invalid_Params( void )
+{
+    MQTTAgentContext_t mqttAgentContext;
+    AgentMessageInterface_t msgInterface = { 0 };
+    MQTTFixedBuffer_t networkBuffer = { 0 };
+    TransportInterface_t transportInterface = { 0 };
+    IncomingPublishCallback_t incomingCallback;
+    void * incomingPacketContext;
+    AgentMessageContext_t msg;
+    MQTTStatus_t mqttStatus;
+
+    /* Check that MQTTBadParameter is returned if any NULL parameters are passed. */
+    mqttStatus = MQTTAgent_Init( NULL, &msgInterface, &networkBuffer, &transportInterface, getTime, incomingCallback, incomingPacketContext );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    mqttStatus = MQTTAgent_Init( &mqttAgentContext, NULL, &networkBuffer, &transportInterface, getTime, incomingCallback, incomingPacketContext );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    mqttStatus = MQTTAgent_Init( &mqttAgentContext, &msgInterface, &networkBuffer, NULL, getTime, incomingCallback, incomingPacketContext );
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+
+    MQTT_Init_ExpectAnyArgsAndReturn( MQTTBadParameter );
```

Comment:
A comment would be helpful
```suggestion
    /* MQTT_Init() should check the network buffer. */
    MQTT_Init_ExpectAnyArgsAndReturn( MQTTBadParameter );
```